controller/lib/hal/hal.h

```diff
@@ -244,22 +260,26 @@ class HalApi {
 private:
   // Initializes watchdog, is called by HalApi::init
   void watchdog_init();
+
+  void setDigitalPinMode(PwmPin pin, PinMode mode);
+  void setDigitalPinMode(BinaryPin pin, PinMode mode);
+
 #ifdef TEST_MODE
   // Instance variables used when mocking HAL.
 
   uint32_t millis_ = 0;
 
-  std::map<AnalogPin, int> analog_pin_values_;
-
-  VoltageLevel digital_pin_values_[14] = {VoltageLevel::HAL_LOW};
-  // The default pin mode on Arduino is INPUT.
+  // The default pin mode on Arduino is INPUT, which happens to be the first
+  // enumerator in PinMode and so the default in these maps!
```

Comment:
Please make sure there's a comment on PinMode::INPUT saying that it has to stay the first, or change the code so it doesn't depend on that. E.g. what if one day we add PinMode::UNSPECIFIED or something as the first value.